src/api/providers/bedrock.ts

```diff
@@ -1,112 +1,155 @@
-		let modelId: string
-		if (this.options.awsUseCrossRegionInference) {
-			let regionPrefix = (this.options.awsRegion || "").slice(0, 3)
-			switch (regionPrefix) {
-				case "us-":
-					modelId = `us.${this.getModel().id}`
-					break
-				case "eu-":
-					modelId = `eu.${this.getModel().id}`
-					break
-				default:
-					// cross region inference is not supported in this region, falling back to default model
-					modelId = this.getModel().id
-					break
-			}
-		} else {
-			modelId = this.getModel().id
-		}
-
-		const stream = await this.client.messages.create({
-			model: modelId,
-			max_tokens: this.getModel().info.maxTokens || 8192,
-			temperature: 0,
-			system: systemPrompt,
-			messages,
-			stream: true,
-		})
-		for await (const chunk of stream) {
-			switch (chunk.type) {
-				case "message_start":
-					const usage = chunk.message.usage
-					yield {
-						type: "usage",
-						inputTokens: usage.input_tokens || 0,
-						outputTokens: usage.output_tokens || 0,
-					}
-					break
-				case "message_delta":
-					yield {
-						type: "usage",
-						inputTokens: 0,
-						outputTokens: chunk.usage.output_tokens || 0,
-					}
-					break
-
-				case "content_block_start":
-					switch (chunk.content_block.type) {
-						case "text":
-							if (chunk.index > 0) {
-								yield {
-									type: "text",
-									text: "\n",
-								}
-							}
-							yield {
-								type: "text",
-								text: chunk.content_block.text,
-							}
-							break
-					}
-					break
-				case "content_block_delta":
-					switch (chunk.delta.type) {
-						case "text_delta":
-							yield {
-								type: "text",
-								text: chunk.delta.text,
-							}
-							break
-					}
-					break
-			}
-		}
-	}
-
-	getModel(): { id: BedrockModelId; info: ModelInfo } {
-		const modelId = this.options.apiModelId
-		if (modelId && modelId in bedrockModels) {
-			const id = modelId as BedrockModelId
-			return { id, info: bedrockModels[id] }
-		}
-		return { id: bedrockDefaultModelId, info: bedrockModels[bedrockDefaultModelId] }
-	}
+    private options: ApiHandlerOptions
+    private client: BedrockRuntimeClient
+
+    constructor(options: ApiHandlerOptions) {
+        this.options = options
+        
+        // Only include credentials if they actually exist
+        const clientConfig: any = {
+            region: this.options.awsRegion || "us-east-1"
+        }
+
+        if (this.options.awsAccessKey && this.options.awsSecretKey) {
+            clientConfig.credentials = {
+                accessKeyId: this.options.awsAccessKey,
+                secretAccessKey: this.options.awsSecretKey
+            }
+            
+            // Only add sessionToken if it exists
+            if (this.options.awsSessionToken) {
+                clientConfig.credentials.sessionToken = this.options.awsSessionToken
+            }
+        }
+
+        this.client = new BedrockRuntimeClient(clientConfig)
+    }
```

Comment:
Will this flow only work if the user or Roo-Cline provides an AWS token?

Reply:
if not provided thre is an error that creentials wren't provid. same as it was before, I think.

Reply:
ah, yup, I see now; thanks!